Lib/zipfile/__init__.py

```diff
@@ -373,6 +377,12 @@ def _sanitize_filename(filename):
     return filename
 
 
+class PreserveMode(enum.Enum):
+    """Options for preserving file permissions upon extraction."""
+    NONE = enum.auto()
+    SAFE = enum.auto()
+    ALL = enum.auto()
```

Comment:
There needs to be a mode for `x` bits only - see https://github.com/python/cpython/issues/59999#issuecomment-2895071926 . (Also this is the behaviour I want and how I found this issue.)

It is not very natural to use an enum here - much more general is to define `preserve_mode` taking an `int` so that I can pass `0o333` or something to preserve `wx` bits and default the `r` bit to `1` (from a default of `0o666`).

The constants here could directly be

```python
PRESERVE_NONE = 0
PRESERVE_SAFE = 0o0777
PRESERVE_ALL = 0o7777
```